packages/arb-token-bridge-ui/tests/e2e/specs/txHistory.cy.ts

```diff
@@ -0,0 +1,137 @@
+import { startWebApp } from '../../support/common'
+
+const DEPOSIT_ROW_IDENTIFIER = /deposit-row-*/i
+const DEPOSIT_SEARCH_IDENTIFIER = /Search for a full or partial L1 tx ID/i
+const WITHDRAWAL_ROW_IDENTIFIER = /withdrawal-row-*/i
+const WITHDRAWAL_SEARCH_IDENTIFIER = /Search for a full or partial L2 tx ID/i
+
+describe('Transaction History', () => {
+  before(() => {
+    cy.changeMetamaskNetwork('goerli')
+    startWebApp()
+  })
+
+  after(() => {
+    cy.logout()
+  })
+
+  // Open tx history panel
+  it('should open transactions history panel', () => {
+    cy.openTransactionsPanel()
+    cy.findByText('Transaction History').should('be.visible')
+  })
+
+  // Test deposits are loading
+  it('should load deposits', () => {
+    cy.findByRole('tab', { name: 'show deposit transactions' })
+      .should('be.visible')
+      .click()
+      .should('have.class', 'selected')
+
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test deposits pagination works
+  it('should paginate deposits', () => {
```

Comment:
Maybe we can add tests for first page (prev button should be disabled) and last page (next button should be disabled)

Reply:
The first case makes sense. Done.
For the second case, we will need to iterate over possibly 20-30 pages before actually being able to test the next page button disabled case. Doesn't seem worth it IMHO.

Reply:
If the user can't go to the last page easily, that's a bit of an UX issue. Might be worth bringing up during our FE meet?